### DISCUSSION TITLE
Abstract out Examine

### DISCUSSION THREAD

**Discussion by bielu** [umbraco/Umbraco-CMS#12493] [Features and ideas]
On one of the meetups, I mentioned I started looking into abstracting out Examine from Umbraco to allow a direct swap to any other search provider.  After noticing there is too many files changed I gave up the idea, but lately I have had plenty of examine issues with hosting in azure and I would love to be able to replace examine without the hacky way how we did in Umbraco v8, so by implementing examine provider for elastic, it was working ok but a lot of responsibility of CMS was built-in into indexer directly.

**Comment by mzajkowski** [comment]
+1000000000000 :) 

It'd be awesome to be able to swap the provider in general. @bielu do you think that the work you already put onto it is completely dropped now? Or is there a chance to do a draft PR even to share the concept / steps performed already?

**Reply by bielu** [reply]
@mzajkowski based on changes in v10 branch I would prefer redone that pr as there would be plenty of merge conflicts. I might draft pr when will have spare time
